src/server/controllers/getPhotoMemory.js

```diff
@@ -0,0 +1,23 @@
+const aws = require('aws-sdk');
+require('env2')('./config.env');
+
+const S3_BUCKET_NAME = process.env.S3_BUCKET_NAME;
+
+module.exports = (req, res) => {
+  aws.config.update({ accessKeyId: process.env.AWSAccessKeyId, secretAccessKey: process.env.AWSSecretKey, signatureVersion: 'v4', region: 'eu-west-2' });
+  const s3 = new aws.S3();
+
+  const s3Params = {
+    Bucket: S3_BUCKET_NAME,
+    Key: req.query.memoryUrl,
```

Comment:
Okay,

  1. This is the s3 key, you should not give this to the client.
  2. The client should send the memoryId
  3. Use the memoryId to find the s3 key
  4. **Check that this memory belongs to the user**. Ie query the memories against the memoryId **and** the userId, to retrieve the s3Key